generator/osm2type.cpp

```diff
@@ -961,42 +1139,103 @@ void PostprocessElement(OsmElement * p, FeatureBuilderParams & params)
 }  // namespace
 
 void GetNameAndType(OsmElement * p, FeatureBuilderParams & params,
-                    function<bool(uint32_t)> const & filterType)
+                    TypesFilterFnT const & filterType, CalculateOriginFnT const & calcOrg)
 {
   // At this point, some preprocessing could've been done to the tags already
   // in TranslatorInterface::Preprocess(), e.g. converting tags according to replaced_tags.txt.
 
   // Stage1: Preprocess tags.
-  PreprocessElement(p);
+  PreprocessElement(p, calcOrg);
 
   // Stage2: Process feature name on all languages.
   NamesExtractor namesExtractor(params);
   ForEachTag(p, namesExtractor);
   namesExtractor.Finish();
 
   // Stage3: Process base feature tags.
-  TagProcessor(p).ApplyRules<void(string &, string &)>({
-      {"addr:housenumber", "*",
-       [&params](string & k, string & v) {
-         params.AddHouseName(v);
-         k.clear();
-         v.clear();
-       }},
-      {"addr:housename", "*",
-       [&params](string & k, string & v) {
-         params.AddHouseName(v);
-         k.clear();
-         v.clear();
-       }},
-      {"addr:street", "*",
-       [&params](string & k, string & v) {
-         params.AddStreet(v);
-         k.clear();
-         v.clear();
-       }},
-      {"addr:postcode", "*",
-       [&params](string & k, string & v) {
-         params.AddPostcode(v);
+  std::string houseName, houseNumber, conscriptionHN, streetHN, addrPostcode;
+  feature::AddressData addr;
+  TagProcessor(p).ApplyRules<void(string &, string &)>(
+  {
+      {"addr:housenumber", "*", [&houseNumber](string & k, string & v)
+      {
+        houseNumber = std::move(v);
+        k.clear();
+        v.clear();
```

Comment:
Can clear be done in ApplyRules?

Reply:
ApplyRules is called in other places so decided not to change. Maybe I can make it a little better.